### DISCUSSION TITLE
Better error handling for resources

### DISCUSSION THREAD

**Discussion by elite174** [solidjs/solid#926] [Ideas]
Hi folks!

Resources are great but there's one inconvenience I noticed about them.

I have this code. I want the `imageDataResource` has proper error state, but if I don't write try/catch block it will crash the app (yes, in this case it's possible to use ErrorBoundary, but I can't use it in my case, because it will just swap the components and I won't be able to process the error. That's why I added extra `error` signal. Can we make it better?
```
const [hasError, setHasError] = createSignal(false);

const processImage = async (file: Nullable<File>): Promise<Nullable<ImageData>> => {
    setHasError(false);

    if (!file || !checkImage(file)) {
      setHasError(true);
      return null;
    }

    try {
      return await uploadImage(file);
    } catch (e) {
      setHasError(true);
      return null;
    }
  };

  const [imageDataResource] = createResource<Nullable<ImageData>, Nullable<File>>(fileToUpload, processImage, {
    initialValue: null,
  });
  ```

**Comment by ConsoleTVs** [comment]
Why is it impossible to use the provided `imageDataResource.error` getter?

**Reply by elite174** [reply]
It's possible
The point is to not fallback to the error boundary
